#!/usr/bin/env python
PACKAGE = "delta_manipulator"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("v_traj", double_t, 0, "trajectory_speed",    0.2, 0.0,   1.0)
gen.add("r_traj", double_t, 0, "trajectory_radius",    0.8, 0.0,   1.0)
gen.add("h_traj", double_t, 0, "trajectory_altitude",    1.0, 0.0,   3.0)


mode_enum = gen.enum([gen.const("static", int_t, 0, "static position hold"),
                 gen.const("line_z", int_t, 1, "line trajectory in z"),
                 gen.const("line_y", int_t, 2, "line trajectory in y"),
                 gen.const("line_x", int_t, 3, "line trajectory in x"),
                 gen.const("circle_xy", int_t, 4, "circular trajectory"),
                 gen.const("figure_8", int_t, 5, "figure_8 trajectory")],
                 "select trajectory type")

gen.add("drone_mode", int_t, 0, "mode selector", 0, 0, 10, edit_method=mode_enum)

exit(gen.generate(PACKAGE, "delta_manipulator", "Trajectory"))